### PR TITLE
v2 Database Migration hsql -> h2

### PR DESCRIPTION
**Pull request by YCCertifai**
# Description
Changes made:
- jdbc driver and some queries (hsql -> h2)
- aspect ratio fix images
- database migration (happens automatically)

This is what I've checked on linux and windows (need help in validation on mac):
- functionalities of classifai (should be no breakdown) 
- aspect ratio fix, can try on this [image](https://drive.google.com/drive/folders/1tPOXrJjfDXYkzED4Zkm2Ch23Q5U1PuJC?usp=sharing)
- old database is completely moved into *.classifai/.archive* folder 
- 3 json files (portfolio, segmentation, boundingbox) are completely moved into *.classifai/.archive* folder
- no data lost from old database
- if error happens during migration ( mainly due to connection failure to any database ), a backup of data will be in *.classifai/.archive* folder, but an empty H2 database will be created. So user can choose to seek for help to retrieve data or move on with clean database. Nevertheless, there will be an error message about that. A document on data retrieving can be prepared if required.

**Feel free to do anything to break it**
But one case is not allowed, which is keep switching v1 and v2 back and forth. This is to prevent the database index get messed up. 
FYI, the database migration only happens when hsql exists, and h2 not exists.
If there is any abnormal behavior please let me know

@codenamewei there is one thing need your advice :
- there is a new column "date" required a default value for old data, at the moment I put it as the time when migration happens.

Please do ask me question if there is any.
Thanks

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Tested on?

- [x] Windows  
- [x] Linux Ubuntu 
- [x] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Comment by YCCertifai** [comment]
Bugs and Code Smells are solved,
Can try to test it now...
Thanks.

**Comment by devenyantis** [comment]
@YCCertifai 

Tested on Windows and Ubuntu18.04
- functionalities of classifai working
- Old data copied to `.classifai/.archive` when database fail to connect. 

Screnshot of database failure.
sum_v2_dbMigration_1

**Comment by YCCertifai** [comment]
@devenyantis for all the v1 distro, the username is SA. Only for cw_v2 brach the username is admin.
Most probably ur previous database is from cw_v2 brach. Try to delete the database, install with any v1 database first then proceed with the test again.

**Comment by devenyantis** [comment]
@YCCertifai 

Database from v1 hsqldb successfully migrate to h2.

**Review by devenyantis — APPROVED** [review]

**Comment by codenamewei** [comment]
@devenyantis 
a little context in there:

reason why we have to put it as admin for v2 onwards is because only when in admin state, [HSQLDB database engine](https://sourceforge.net/projects/hsqldb/files/) of h2/hsqldb can be connected to the database to do queries from. 
meaning there is this gui where you can connect to database and do queries in the window

we failed to do that in v1 and do not implement the changes as it will be backward incompatible for existing user who already using v1 (upon installing new version)

hence for v1 and v2, the database cant be the same.

i always rename my folder .classifai_v1 .classifai_v2 when swapping v1 and v2 version

**Comment by codenamewei** [comment]
ill take a look at this PR this week. kindly on hold merging of this branch thanks.

meanwhile, if you plan to do development on v2, feel free to create a new branch based on this branch, the conflict of changes will likely be smaller upon merging

```
git checkout -b new_branch base_branch

Example:
git checkout -b cw_branch sum_v2_dbMigration
```

thanks guys
@devenyantis @YCCertifai

File: classifai-core/src/main/java/ai/classifai/database/DatabaseConfig.java [hunk 1]
```diff
@@ -1,37 +1,21 @@
-/*
- * Copyright (c) 2020-2021 CertifAI Sdn. Bhd.
- *
- * This program and the accompanying materials are made available under the
- * terms of the Apache License, Version 2.0 which is available at
- * https://www.apache.org/licenses/LICENSE-2.0.
- *
- * Unless required by applicable law or agreed to in writing, software
```
Comment (codenamewei) [inline classifai-core/src/main/java/ai/classifai/database/DatabaseConfig.java:8]: @YCCertifai license header missing

**Review by codenamewei — CHANGES_REQUESTED** [review]

File: classifai-core/src/main/java/ai/classifai/database/DatabaseConfig.java (hunk 1 above)
Comment (YCCertifai) [reply classifai-core/src/main/java/ai/classifai/database/DatabaseConfig.java]: fixed it.. I wonder when I delete it .....

**Comment by YCCertifai** [comment]
@codenamewei 
changes made:
1. final variable -> all capital letters separated with "_"
2. functios rename -> less verbose name
3. spacing consistency
4. --unlockdb arg is deleted, with modified comment
5. intermediate json files are deleted upon migration
6. logging message review.
7. If duplicated access of databases/classifai, a very wild error message will generated in the log file, stating that unable to connect to database. 
Screenshot from 2021-02-03 02-26-27
8. I fixed the major code smells, but the minor are all naming convention, sonarcloud doesn't like full capital letters...

**Review by codenamewei — APPROVED** [review]

**Pull request merged**